File: ex2/ex2.ts
import * as crypto from "crypto";

const puzzleID = 'BBSE_E01';
const d =  '0000';
var x = 0;
var result;

while (true) {
    result = puzzleID + String(x);
    result = crypto.createHash('sha256').update(result).digest('hex');
    
    if (result.substr(0,4) == d) {
        break
    }

    x += 1
}

console.log('Found x' + String(x) + ' returns hash value ' + String(result));

File: ex2/ex2.py
import hashlib

puzzleID = 'BBSE_E01'
d = '0000'
x = 0

while True:
    result = puzzleID + str(x)
    result = result.encode()
    result = hashlib.sha256(result).hexdigest()
    if result[:4] == d:
        break
    x += 1

print('Found x' + str(x) + ' returns hash value ' + str(result))